README.md

```diff
@@ -1,22 +1,123 @@
 
+[url-npm]: https://www.npmjs.com/package/generator-swiftserver
+
 This module is a [Yeoman](http://yeoman.io) generator for creating REST webservices based on the [Kitura web framework](http://kitura.io) with the [Swift](https://swift.org/) language.
 
 This generator and the Yeoman library runs on Node.js and generates a Swift 4 application.
 
-## Installation
+## Contents
+1. [Prerequisites](#prerequisites)
+1. [Installation](#installation)
+1. [Usage](#usage)
+    - [Options](#options)
+    - [Run](#run)
+1. [Generated Artifacts](#generated-artifacts)
+1. [Development](#development)
+1. [Testing](#testing)
+1. [Contributing](#contributing)
+
+## Prerequisites
 To use this module, you will need Node.js and Swift 4 installed on your Linux or macOS system. You can get Node.js from https://nodejs.org and Swift 4 from https://swift.org/download.
 
-You will also need the Yeoman command line utility [yo](https://github.com/yeoman/yo) installed in your global Node.js module directory.  
-To install yo, use: `npm install -g yo`  
-To install this generator, use: `npm install -g git+https://github.com/IBM-Swift/generator-swiftserver`
+You will also need the Yeoman command line utility [yo](https://github.com/yeoman/yo) installed in your global Node.js module directory.
+ 
+To install `yo`, run:
+
+```bash
+npm install -g yo
+```
+
+## Installation
+To install generator-swiftserver, run:
+
+```bash
+npm install -g git+https://github.com/IBM-Swift/generator-swiftserver
+```
+
+## Usage
+To create a Swift Server Generator project with no models defined, run:
+
+```bash
+yo swiftserver
+```
+
+To add a data model to your generated project, `cd` to the new project directory and run:
 
-## Quick start
-To create a Swift Server Generator project with no models defined, use: `yo swiftserver`  
-To add a data model to your project, `cd` to the new project directory and use: `yo swiftserver:model`  
+```bash
+yo swiftserver:<model>
+```
+
+### Options
+You can also change how generator-swiftserver is invoked by choosing options (in below table):
+
+```bash
+yo swiftserver [options]
+```
+
+Option | Description
+--- | ---
+`--init` | Generate basic default scaffold without prompting user for input.
+`--skip-build` | Skip building the generated application
+`--single-shot` | Creates application without including generator metadata files
+`--bluemix` | Invoke generator-swiftserver using a bluemix json object
+`--name` | Project name
+`--type` | Give a specific type of application to generate. (`web`, `bff`, `scaffold`, `crud`)
+`--metrics` | Generate embedded metrics dashboard for project
+`--docker` | Generate Dockerfile for project
+`--healthcheck` | Add health checking to project
+`--spec` | Invoke generator-swiftserver using a spec json object
+
+### Run
 To run the generated server, use: `<project-dir>/.build/debug/<app-name>`
+
+## Generated Artifacts
+
+File | Purpose
+--- | ---
+Dockerfile | Configuration file for the run container
```

Comment:
I would mention the Cloud Enablement generator here for these specialized files, but I am open to suggestions.